src/helpers/qownspellchecker.cpp

```diff
@@ -27,9 +27,12 @@ QOwnSpellChecker::QOwnSpellChecker(QObject *parent) : QObject(parent) {
     } else {
         autoDetect = false;
     }
+
 #ifdef Q_OS_MACOS
-    QString s = spellchecker->availableLanguages().at(0);
-    spellchecker->setDefaultLanguage(s);
+    QStringList s = spellchecker->availableLanguages();
+    if (!s.contains(spellchecker->defaultLanguage())) {
+        spellchecker->setDefaultLanguage(spellchecker->availableLanguages().at(0));
```

Comment:
Crashes at startup here, I guess because no dictionaries are available by default.

Reply:
But it also crashes if I put dictionaries into `~/Library/Spelling`.

```
Debug:  HunspellClient::HunspellClient
Warning: No language dictionaries for the language: "da" (../src/libraries/sonnet/src/core/loader.cpp:101, Sonnet::SpellerPlugin *Sonnet::Loader::createSpeller(const QString &, const QString &) const)
Fatal: ASSERT failure in QList<T>::at: "index out of range", file /Applications/Qt/5.13.0/clang_64/lib/QtCore.framework/Headers/qlist.h, line 560 (/Applications/Qt/5.13.0/clang_64/lib/QtCore.framework/Headers/qlist.h:560, (null))
```

Reply:
is it detecting the dictionaries?

Reply:
No. The list is empty.